### PR TITLE
9051 Effective Date Time component

### PR DESCRIPTION
**Pull request by severinbeauvais**
*Issue #:* /bcgov/entity#9051

*Description of changes:*
- deleted unnecessary package lock files (revert from previous PR)
- EffectiveDateTime component 
- unit tests
- updated Storybook

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the business-filings-ui license (Apache 2.0).

### CODE REVIEW

File: src/components/EffectiveDateTime/EffectiveDateTime.vue [hunk 1]
```diff
@@ -0,0 +1,477 @@
...
+          </p>
+        </v-col>
+      </v-row>
+    </v-form>
+  </v-card>
+</template>
+
+<script lang="ts">
+import { Component, Emit, Mixins, Prop, Vue, Watch } from 'vue-property-decorator'
+import { DatePicker } from '@/components/DatePicker' // NB: local component
```
Comment (severinbeauvais) [inline src/components/EffectiveDateTime/EffectiveDateTime.vue:77]: Ideally we would import the bcrs-shared-component DatePicker here, but I was getting Storybook errors here (while everything else worked).
I killed a few hours on this so ... this is what I'm going with atm.

File: src/components/FeeSummary/package-lock.json [hunk 2]
```diff
@@ -1,23 +0,0 @@
-{
```
Comment (severinbeauvais) [inline src/components/FeeSummary/package-lock.json:1]: It turns out we probably don't need these package lock files after all.

Oddly, "lerna hoist" created these files once when I ran it. But then I opened up a new window and ran everything from scratch, they weren't created (and everything still worked).

So, I've deleted all these now :)

File: src/components/index.ts [hunk 3]
```diff
@@ -0,0 +1,2 @@
+// local component
+export { DatePicker } from './DatePicker'
```
Comment (severinbeauvais) [inline src/components/index.ts:2]: This is here so that Storybook can find the "local" DatePicker component. The app needs to do the same thing (but there, DatePicker will be an external component).

This is because Storybook can't find `@bcrs-shared-components/DatePicker` ... I don't know why. Lots of time spent on this already :(  This works and it's not terribly ugly.

File: src/components/EffectiveDateTime/EffectiveDateTime.vue [hunk 4]
```diff
@@ -0,0 +1,477 @@
...
+  position: relative;
+
+  .validation-alert-msg {
+    line-height: 12px;
+    position: absolute;
+    top: -2rem;
+    padding: 0 12px;
+    font-size: 12px;
+    font-weight: 500;
+    color: $BCgovInputError !important;
```
Comment (cameron-freshworks) [inline src/components/EffectiveDateTime/EffectiveDateTime.vue:463]: Are these defined in the shared components?

File: src/components/EffectiveDateTime/EffectiveDateTime.vue [hunk 5]
```diff
@@ -0,0 +1,477 @@
...
+      !this.isUnderTime &&
+      !this.isOverTime &&
+      validDateText
+    )
+  }
+}
+</script>
+
+<style lang="scss" scoped>
+@import '@/assets/styles/theme.scss';
```
Comment (cameron-freshworks) [inline src/components/EffectiveDateTime/EffectiveDateTime.vue:405]: Related to the other question, is this something we have defined here? or could this present an issue if pulled in and use in a project without this hierarchy.

Comment (severinbeauvais) [reply src/components/EffectiveDateTime/EffectiveDateTime.vue:405]: Yes, if a parent project is missing this folder/file, this will fail. But it has always been this way for all our projects.

All our projects (except NR UI) use the folders above.

Comment (cameron-freshworks) [reply src/components/EffectiveDateTime/EffectiveDateTime.vue:405]: Ok cool

File: src/components/EffectiveDateTime/EffectiveDateTime.vue (hunk 4 above)
Comment (severinbeauvais) [reply src/components/EffectiveDateTime/EffectiveDateTime.vue:463]: Yes, these CSS variables are defined in https://github.com/bcgov/bcrs-shared-components/blob/main/src/assets/styles/theme.scss.

And yes, the parent app needs to have these defined as well.

I think/hope the build steps will identify any incorrect imports/variables.

**Review by cameron-freshworks — APPROVED** [review]
Looks good!

File: src/components/EffectiveDateTime/EffectiveDateTime.vue (hunk 4 above)
Comment (cameron-freshworks) [reply src/components/EffectiveDateTime/EffectiveDateTime.vue:463]: Copy that!

File: src/components/EffectiveDateTime/EffectiveDateTime.vue (hunk 1 above)
Comment (severinbeauvais) [reply src/components/EffectiveDateTime/EffectiveDateTime.vue]: I updated this. See comment on `src/components/index.ts` below.

TLDR: this line is now more obviously a "local import" and it looks the same in the parent app; but the `components/index.ts` file will reference either the local file or the external file, respectively. This keeps things generic/consistent as much as possible while still providing the needed references.

**Review by eve-git — APPROVED** [review]
Nice!
